src/settings.cmake

```diff
@@ -189,8 +189,39 @@ if(WIN32)
     set(CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")
     set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO} /NODEFAULTLIB:libucrt.lib /DEFAULTLIB:ucrt.lib")
 else()
+    add_compile_options(-Wall)
+    add_compile_options(-Wextra)
+    if(CMAKE_C_COMPILER_ID STREQUAL Clang)
+        add_compile_options(-Weverything)
+    endif()
+    add_compile_options(-Werror)
+    add_compile_options(-Wno-c++98-compat)
+    add_compile_options(-Wno-c++98-compat-pedantic)
```

Comment:
What's the plan for all of these? Is it a point in time thing that you think should get fixed eventually?
Based on the CLang discussion, I think it would make sense to not use `-Weverything`.
CoreCLR is definitely not using that: https://github.com/dotnet/coreclr/blob/e6034d903f2608445a3f66e3694f461fad7b8b88/configurecompiler.cmake#L420

I'm not arguing against fixing warnings, but if we have to disable so many, maybe we should not be turning them all on by default.

Reply:
I am going to keep working on this for a bit; once I hit a wall, I will see about turning off -Weverything. If that still results in disabling too many warnings, we can start selectively enabling warnings instead of turning on everything. What do you think?

Reply:
If we have trouble making `-Wall` alone working... we'll have to reconcider, hopefully not. Having `Wextra` working would be nice eventually.

I'm a little bit worried about Windows - it's a different compiler... I just hope that there are no warnings which go against each other.